cmake/BuildType.cmake

```diff
@@ -49,7 +49,7 @@ if(CMAKE_COMPILER_IS_GNUCC)
     set(GCC_OPTFLAGS -O2)
   endif()
 
-  set(CMAKE_C_FLAGS_DEBUG "-DO_DEBUG -DO_DEBUG_ATOMGC -O0 ${GCC_GFLAGS} $ENV{CFLAGS}"
+  set(CMAKE_C_FLAGS_DEBUG "-DO_DEBUG -DO_DEBUG_ATOMGC -O0 ${GCC_GFLAGS} $ENV{CFLAGS} -DO_CHECK_TERM_REFS -DSECURE_GC -DCHK_SECURE -DO_MAINTENANCE -gdwarf-2 -g3"
```

Comment:
You might not agree with this change -- but it has the flags that caused https://github.com/SWI-Prolog/swipl-devel/issues/1292 when doing a debug build.

Reply:
Skipped this :smile: